eslintPlugin/src/noAnimatedStyleToNonAnimatedComponent.ts

```diff
@@ -0,0 +1,175 @@
+import { AST_NODE_TYPES, TSESLint, TSESTree } from '@typescript-eslint/utils';
+
+const rule: TSESLint.RuleModule<'animatedStyle' | 'sharedValue'> = {
+  create: function (context) {
+    return {
+      JSXOpeningElement(node: TSESTree.JSXOpeningElement) {
+        if (node.name.type === AST_NODE_TYPES.JSXMemberExpression) return; // Property-like namespace syntax <Animated.View>
+        if (node.name.type === AST_NODE_TYPES.JSXNamespacedName) return; // XML-based namespace syntax: <Animated:View>
+
+        const sourceCode = context.getSourceCode();
+        const tokensBefore = sourceCode.getTokensBefore(node);
+        const componentName = node?.name?.name;
+
+        main();
+        function main() {
+          if (
+            isVariableDefinedAs(componentName, 'Animated') || //TODO allow to customize rule and provide custom import name
+            isVariableDefinedAs(componentName, 'Reanimated') ||
+            isVariableDefinedAs(componentName, 'createAnimatedComponent')
+          ) {
+            return;
+          }
+          const styleAttribute = node.attributes
+            .map((attribute) => {
+              return attribute.type === AST_NODE_TYPES.JSXAttribute &&
+                attribute.name.name === 'style'
+                ? [attribute]
+                : [];
+            })
+            .flat();
```

Comment:
How about using `.find` method here instead of `.map`?

Reply:
Oh, this is actually a trick here ;-)
I've tried with `.find` first and for some reason it require you to make a type case:
<img width="726" alt="image" src="https://github.com/software-mansion/react-native-reanimated/assets/56199675/28a5fd98-9844-4ae9-a31b-26847037eb61">
But with map you get the correct type
<img width="726" alt="image" src="https://github.com/software-mansion/react-native-reanimated/assets/56199675/bdd6136d-1da1-4dd5-bff6-5e471e2f85e0">

Reply:
This is what type assertions are for. 

You shouldn't use `map` and `flat` just to get a type _you want_. You should use `find` and type assertion if you know better than TypeScript.

What you are doing at the moment is an actual type assertion but with unnecessary runtime decisions and I believe this is a bad pattern.

Reply:
@tjzel Yes, you're right, in this case I really know the actual type.